glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -2337,6 +2453,58 @@ where
         }
     }
 
+    // The function is used to send an AUTH command to a node and then retry the original request.
+    // The function is used when the original request failed due to an AUTH error.
+    // Cases in which the function is helpful:
+    // 1. The password was changed, the connection was not re-established, and the request failed due to an AUTH error.
+    // 2. The protection method is allowing X hours of connection without authentication, and the request failed due to an AUTH error.
+    async fn send_re_auth_request(
+        core: Core<C>,
+        info: RequestInfo<C>,
+        address: String,
+    ) -> OperationResult {
+        let password = core
+            .get_cluster_param(|params| params.password.clone())
+            .expect(MUTEX_READ_ERR);
+        let username = core.get_cluster_param(|params| params.username.clone());
+        let Some(password) = password else {
+            return Err((
+                OperationTarget::Node { address },
+                RedisError::from((
+                    ErrorKind::AuthenticationFailed,
+                    "No password provided for AUTH",
+                )),
+            ));
+        };
+        let mut auth_cmd = crate::cmd("AUTH");
+        if let Ok(Some(username)) = username {
+            auth_cmd.arg(username);
+        }
+        auth_cmd.arg(password);
+        let cmd = Arc::new(auth_cmd.to_owned());
+        let routing =
+            InternalRoutingInfo::SingleNode(InternalSingleNodeRouting::ByAddress(address.clone()));
+        let auth_info = RequestInfo {
+            cmd: CmdArg::Cmd { cmd, routing },
+        };
+        let response = Self::try_request(auth_info, core.clone()).await;
+        match response {
+            Ok(response) => {
+                if let Response::Single(Value::Okay) = response {
+                    Self::try_request(info, core.clone()).await
+                } else {
```

Comment:
we might be missing here unexpected failures - please catch the exception and add it to the error message. and/or create a different error if we get a response which isn't an error not Value::Okay, (for example lets say the valkey 8.5 would change the response to something else, it would be hard to find out why our code breaks as we hide the response here)